core/pom.xml

```diff
@@ -305,7 +309,8 @@
                 <version>0.10_1.5.1</version>
                 <configuration>
                     <sourceDirectories> <!-- (Optional) Paths to source-directories. Overrides ${project.build.sourceDirectory} -->
-                        <param>${project.basedir}/src/main/scala</param>
+                        <sourceDirectory>${project.basedir}/src/main/scala</sourceDirectory>
+                        <sourceDirectory>${project.basedir}/src/main/${profile.source.directory}</sourceDirectory>
```

Comment:
shall we add both folder `spark-2.3` and `spark-2.4`?

Reply:
I'm not sure of that. Both implementation is reasonable.

Reply:
we should format all the source code

Reply:
so that we shall hardcode all available profiles, and remember to change them when new profiles are provided/modified.

Reply:
or ci `/build` command will trigger both `-Pspark2.3` and `-Pspark2.4` profile

Reply:
It is!